File: components_page/components/modal/centered.R
library(dashBootstrapComponents)
library(dashHtmlComponents)

modal_centered <- htmlDiv(
  list(
    dbcButton("Open", id = "open-centered", n_clicks = 0),
    dbcModal(
      list(
        dbcModalHeader("Header"),
        dbcModalBody("This modal is vertically centered"),
        dbcModalFooter(
          dbcButton(
            "Close", id = "close-centered", n_clicks = 0, className = "ml-auto"
          )
        )
      ),
      id = "modal-centered",
      is_open = FALSE,
      centered = TRUE
    )
  )
)


app$callback(
  output("modal-centered", "is_open"),
  list(
    input("open-centered", "n_clicks"),
    input("close-centered", "n_clicks"),
    state("modal-centered", "is_open")
  ),
  function(n1, n2, is_open) {
    if (n1 > 0 | n2 > 0) {
      return(!is_open)
    }
    return(is_open)
  }
)

File: docs/components_page/components/modal/centered.py
import dash_bootstrap_components as dbc
from dash import Input, Output, State, html

modal = html.Div(
    [
        dbc.Button("Open", id="open-centered"),
        dbc.Modal(
            [
                dbc.ModalHeader(dbc.ModalTitle("Header"), close_button=True),
                dbc.ModalBody("This modal is vertically centered"),
                dbc.ModalFooter(
                    dbc.Button(
                        "Close",
                        id="close-centered",
                        className="ms-auto",
                        n_clicks=0,
                    )
                ),
            ],
            id="modal-centered",
            centered=True,
            is_open=False,
        ),
    ]
)


@app.callback(
    Output("modal-centered", "is_open"),
    [Input("open-centered", "n_clicks"), Input("close-centered", "n_clicks")],
    [State("modal-centered", "is_open")],
)
def toggle_modal(n1, n2, is_open):
    if n1 or n2:
        return not is_open
    return is_open
